cache/redis/redis.go

```diff
@@ -16,9 +17,33 @@ type Redis struct {
 	client *redis.Client
 }
 
-func Init(ctx context.Context, host string) (*Redis, error) {
+type Option func(o *redis.Options)
+
+// WithTLS options allows to set TLS config to initiate
+// connection with Redis host with TLS transport protocol enabled.
+// Example configuration for tests, self-signed untrusted certs.
+//
+//	&tls.Config{
+// 		InsecureSkipVerify: true,
+//	}
+//
+// Example of configuration for production usage
+//	&tls.Config{
+//		MinVersion: tls.VersionTLS12,
+//	}
+func WithTLS(cfg *tls.Config) Option {
+	return func(o *redis.Options) {
+		o.TLSConfig = cfg
+	}
+}
+
+func Init(ctx context.Context, host string, opts ...Option) (*Redis, error) {
 	options, err := redis.ParseURL(host)
 
+	for _, opt := range opts {
+		opt(options)
```

Comment:
Maybe you want to apply options after error checking

Reply:
Right, somehow missed that